EzMap.Api/Controllers/TagController.cs

```diff
@@ -0,0 +1,109 @@
+﻿using EzMap.Api.Services;
+using EzMap.Domain.Dtos;
+using EzMap.Domain.Repositories;
+using EzMap.Domain.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EzMap.Api.Controllers;
+
+
+[Route("api/[controller]")]
+[ApiController]
+public class TagController : ControllerBase
+{
+    [Authorize]
+    [HttpPost("")]
+    public async Task<IActionResult> Create([FromBody] TagCreateDto dto, [FromServices] IUnitOfWork uow,
+        [FromServices] IIdentityService identityService)
+    {
+        if (string.IsNullOrEmpty(dto.Name)
+            && string.IsNullOrEmpty(dto.Description))
+        {
+            return BadRequest("Kindly fill Name, Description fields!");
+        }
+        
+        uow.TagRepository.AddTag(dto.WithUserId(identityService.GetUserId()));
+
+        if (await uow.SaveAsync() > 0)
+        {
+            return Ok("Your tag is created successfully!");
+        }
+
+        return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+    }
+
+    [Authorize]
+    [HttpPut("{id:guid}")]
+    public async Task<IActionResult> UpdateTag([FromBody] TagUpdateDto dto, [FromServices] IUnitOfWork uow,
+        [FromServices] IIdentityService identityService)
+    {
+        if (string.IsNullOrEmpty(dto.Name)
+            && string.IsNullOrEmpty(dto.Description))
+        {
+            return BadRequest("Kindly fill Name, Description fields!");
+        }
+
+        var dbTag = await uow.TagRepository.GetTagById(identityService.GetUserId(), dto.Id);
+
+        if (dbTag is not null)
+        {
+            uow.TagRepository.UpdateTag(dbTag, dto.WithUserId(identityService.GetUserId()));
+        }
+
+        if (await uow.SaveAsync() > 0) return Ok("Your tag is updated successfully!");
+
+        return new StatusCodeResult(StatusCodes.Status500InternalServerError);
```

Comment:
Can use ternary operator her.

Please make the code uniform. If you can't do that extract the repetitive code to a common method.